extension/src/popup/views/ViewPublicKey/index.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import QrCode from "qrcode.react";
+import CopyToClipboard from "react-copy-to-clipboard";
+import { Formik, Field, FieldProps } from "formik";
+
+import { emitMetric } from "helpers/metrics";
+import { truncatedPublicKey } from "helpers/stellar";
+
+import { Form } from "popup/basics/Forms";
+import { ROUTES } from "popup/constants/routes";
+import { METRIC_NAMES } from "popup/constants/metricsNames";
+import { navigateTo, openTab } from "popup/helpers/navigate";
+import { BackButton } from "popup/basics/Buttons";
+import { PopupWrapper } from "popup/basics/PopupWrapper";
+import { BottomNav } from "popup/components/BottomNav";
+import {
+  accountNameSelector,
+  publicKeySelector,
+  updateAccountName,
+} from "popup/ducks/accountServices";
+import { settingsNetworkDetailsSelector } from "popup/ducks/settings";
+
+import { Button, TextLink, Icon, Input } from "@stellar/design-system";
+
+import "./styles.scss";
+
+export const ViewPublicKey = () => {
+  const publicKey = useSelector(publicKeySelector);
+  const accountName = useSelector(accountNameSelector);
+  const [isCopied, setIsCopied] = useState(false);
+  const [isEditingName, setIsEditingName] = useState(false);
+  const { network } = useSelector(settingsNetworkDetailsSelector);
+
+  const dispatch = useDispatch();
+
+  interface FormValue {
+    accountName: string;
+  }
+
+  const initialValues: FormValue = {
+    accountName,
+  };
+
+  const handleSubmit = async (values: FormValue) => {
+    const { accountName: newAccountName } = values;
+    if (accountName !== newAccountName) {
+      await dispatch(updateAccountName(newAccountName));
+      emitMetric(METRIC_NAMES.viewPublicKeyAccountRenamed);
+    }
+    setIsEditingName(false);
+  };
+
+  const handleCopy = () => {
+    setIsCopied(true);
+    emitMetric(METRIC_NAMES.viewPublicKeyCopy);
+
+    const t = setTimeout(() => {
+      setIsCopied(false);
+      clearTimeout(t);
+    }, 1000);
+  };
+
+  return (
+    <PopupWrapper>
+      <div className="ViewPublicKey">
+        <BackButton isPopup onClick={() => navigateTo(ROUTES.account)} />
+        <div className="ViewPublicKey__header">
+          {isEditingName ? (
+            <Formik initialValues={initialValues} onSubmit={handleSubmit}>
+              <Form className="ViewPublicKey__form">
```

Comment:
I would actually replace this `Form` with `import { Form } from "formik"`;

That way, you won't have to overwrite any styles with `!important` in `&__form`

Reply:
👍  https://github.com/stellar/freighter/pull/375/commits/58945962b1510042c937b5bca05a73fc89b0494e